### PR TITLE
Support for hyperdisk params

### LINKED ISSUE #702: Add Support for Google Cloud Hyperdisk storage in worker volume types
**How to categorize this issue?**

/area storage
/kind enhancement
/platform gcp

**What would you like to be added**:
This request is about the addition of support for Google Cloud's new generation block storage, [Hyperdisk](https://cloud.google.com/compute/docs/disks/hyperdisks), in worker volume types.

**Why is this needed**:
Google Cloud's Hyperdisk offers enhanced performance and scalability compared to traditional block storage options. Integrating support for Hyperdisk into worker would provide users with greater flexibility and performance optimization capabilities for their Kubernetes clusters.

### PR DESCRIPTION
**Pull request by hebelsan**
**How to categorize this PR?**

/area quality
/kind enhancement
/platform gcp

**What this PR does / why we need it**:
Support and validation of disk params provisioned-iops and provisioned-throughput

**Which issue(s) this PR fixes**:
Fixes #702 

**Special notes for your reviewer**:

**Release note**:

```other operator
Support and validation of disk params provisioned-iops and provisioned-throughput
```

### CODE REVIEW

**Comment by hebelsan** [comment]
MCM PR: https://github.com/gardener/machine-controller-manager-provider-gcp/pull/121

**Review by kon-angelo — COMMENTED** [review]
/lgtm

**Comment by hebelsan** [comment]
Needs some adjustment (should be configurable per dataVolume disk) as specified here:
https://github.com/gardener/gardener-extension-provider-gcp/pull/751

File: pkg/controller/worker/machines.go [hunk 1]
```diff
@@ -41,6 +42,15 @@ const (
...
 
+var (
+	// AllowedTypesIops are the volume types for which iops can be configured
+	// https://cloud.google.com/sdk/gcloud/reference/compute/disks/create#--provisioned-iops
+	AllowedTypesIops = []string{"pd-extreme", "hyperdisk-extreme"}
+	// AllowedTypesThroughput are the the volume types for which throughput can be configured
+	// https://cloud.google.com/sdk/gcloud/reference/compute/disks/create#--provisioned-throughput
+	AllowedTypesThroughput = []string{"hyperdisk-throughput"}
+)
+
```
Comment (AndreasBurger) [inline pkg/controller/worker/machines.go:53]: Consider defining these strings in the constants section (and put the consts in the slice)

File: pkg/apis/gcp/validation/shoot.go [hunk 2]
```diff
@@ -57,7 +57,9 @@ func validateVolume(vol *core.Volume, fldPath *field.Path) field.ErrorList {
 	}
 	if vol.Type != nil && *vol.Type == worker.VolumeTypeScratch {
 		allErrs = append(allErrs, field.Invalid(
-			fldPath.Child("type"), worker.VolumeTypeScratch, fmt.Sprintf("type %s is not allowed as boot disk", worker.VolumeTypeScratch)))
+			fldPath.Child("type"),
+			worker.VolumeTypeScratch,
+			fmt.Sprintf("type %s is not allowed as boot disk", worker.VolumeTypeScratch)))
```
Comment (AndreasBurger) [inline pkg/apis/gcp/validation/shoot.go:62]:
```suggestion
			fldPath.Child("type"),
			worker.VolumeTypeScratch,
			fmt.Sprintf("type %s is not allowed as boot disk", worker.VolumeTypeScratch),
		))
```
Just a suggestion, keeps future diffs smaller. Also seems more consistent to me, but I realise that this was already done this way before.

**Review by AndreasBurger — APPROVED** [review]
lgtm

**Review by AndreasBurger — APPROVED** [review]

**Pull request merged**